File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  if (input === input.toUpperCase() && input.search(/[A-z]/) !== -1) {
    return 'Whoa, chill out!';
  } else if (input.charAt(input.length - 1) === '?') {
    return 'Sure.';
  } else if (input.replace(/\s*/, '').length === 0){
    return 'Fine. Be that way!';
  } else {
    return 'Whatever.';
  }
};

module.exports = Bob;

File: ruby/bob/bob.rb
class Bob
  def initialize
  end

  def hey(phrase)
    if yelling?(phrase)
      'Whoa, chill out!'
    elsif question?(phrase)
      'Sure.'
    elsif no_content?(phrase)
      'Fine. Be that way!'
    else
      'Whatever.'
    end
  end

  private

  def yelling?(phrase)
    phrase.upcase == phrase && phrase.upcase.scan(/[A-Z]/).length > 0
  end

  def question?(phrase)
    phrase.chars.last == '?'
  end

  def no_content?(phrase)
    phrase.gsub(/\s+/, '').empty?
  end
end
